Assets/Scripts/CardSystem/DeckManager.cs

```diff
@@ -279,5 +296,86 @@ public void AddTemporaryCardToDeck(List<Card> newCards)
         }
+
+        // ISaveable Implementation
+        public string SaveId => _persistentId ? _persistentId.Id : string.Empty;
+
+        public object CaptureState()
+        {
+            return new DeckSaveData
+            {
+                baseDeckCardNames = baseDeck.Select(card => card ? card.name : "").ToArray(),
+                activeDeckCardNames = activeDeck.Select(card => card ? card.name : "").ToArray(),
+                discardPileCardNames = discardPile.Select(card => card ? card.name : "").ToArray(),
+                slotCardNames = cardManager.cardSlots.Select(card => card ? card.name : "").ToArray()
+            };
+        }
+
+        public void RestoreState(object state)
+        {
+            if (state is DeckSaveData data)
+            {
+                // Rebuild base deck from saved card names
+                baseDeck.Clear();
+                foreach (var cardName in data.baseDeckCardNames)
+                {
+                    if (!string.IsNullOrEmpty(cardName))
+                    {
+                        var card = Resources.LoadAll<Card>("").FirstOrDefault(c => c.name == cardName);
+                        if (card) baseDeck.Add(card);
+                    }
+                }
+
+                // Rebuild active deck from saved card names
+                activeDeck.Clear();
+                foreach (var cardName in data.activeDeckCardNames)
+                {
+                    if (!string.IsNullOrEmpty(cardName))
+                    {
+                        var card = baseDeck.FirstOrDefault(c => c.name == cardName);
+                        if (card) activeDeck.Add(card);
+                    }
+                }
+
+                // Rebuild discard pile from saved card names
+                discardPile.Clear();
+                foreach (var cardName in data.discardPileCardNames)
+                {
+                    if (!string.IsNullOrEmpty(cardName))
+                    {
+                        var card = baseDeck.FirstOrDefault(c => c.name == cardName);
+                        if (card) discardPile.Add(card);
+                    }
+                }
+
+                // Restore cards in slots
+                for (int i = 0; i < data.slotCardNames.Length && i < cardManager.cardSlots.Length; i++)
+                {
+                    if (!string.IsNullOrEmpty(data.slotCardNames[i]))
+                    {
+                        var card = baseDeck.FirstOrDefault(c => c.name == data.slotCardNames[i]);
+                        if (card)
+                        {
+                            cardManager.SetCardInSlot(i, card);
+                        }
+                    }
+                    else
+                    {
+                        cardManager.SetCardInSlot(i, null);
+                    }
+                }
+
+                Debug.Log($"Deck state restored: {activeDeck.Count} active, {discardPile.Count} discarded");
```

Comment:
Figured out the issue.
```suggestion
                // Restore cards in slots
                for (int i = 0; i < data.slotCardNames.Length && i < cardManager.cardSlots.Length; i++)
                {
                    if (!string.IsNullOrEmpty(data.slotCardNames[i]))
                    {
                        var card = baseDeck.FirstOrDefault(c => c.name == data.slotCardNames[i]);
                        if (card)
                        {
                            cardManager.SetCardInSlot(i, card);
                        }
                    }
                }

                Debug.Log($"Deck state restored: {activeDeck.Count} active, {discardPile.Count} discarded");
```

Sending a null card to the `SetCardInSlot` function will cause an error. My assumption is that this would be because there is supposed to be nothing in this slot, so telling it to be null seems wrong.